src/ui/events.js

```diff
@@ -251,32 +292,33 @@ export type MapBoxZoomEvent = {
 };
 
 /**
- * A `MapDataEvent` object is emitted with the {@link Map.event:data}
- * and {@link Map.event:dataloading} events. Possible values for
- * `dataType`s are:
- *
- * - `'source'`: The non-tile data associated with any source
- * - `'style'`: The [style](https://www.mapbox.com/mapbox-gl-style-spec/) used by the map
+ * `MapDataEvent` is the type for objects emitted by data-related [`Map` events](https://docs.mapbox.com/mapbox-gl-js/api/map/#map-events).
  *
  * @typedef {Object} MapDataEvent
- * @property {string} type The event type.
- * @property {string} dataType The type of data that has changed. One of `'source'`, `'style'`.
+ * @property {string} type The type of originating event.
+ * @property {string} dataType The type of data that has changed. One of `'source'` or `'style'`, where `'source'` refers to the non-tile data associated with any source, and `'style'` refers to the [style](https://docs.mapbox.com/help/glossary/style/) used by the map.
  * @property {boolean} [isSourceLoaded] True if the event has a `dataType` of `source` and the source has no outstanding network requests.
- * @property {Object} [source] The [style spec representation of the source](https://www.mapbox.com/mapbox-gl-style-spec/#sources) if the event has a `dataType` of `source`.
+ * @property {Object} [source] The [style spec representation of the source](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/) if the event has a `dataType` of `source`.
  * @property {string} [sourceDataType] Included if the event has a `dataType` of `source` and the event signals
  * that internal data has been received or changed. Possible values are `metadata`, `content` and `visibility`.
  * @property {Object} [tile] The tile being loaded or changed, if the event has a `dataType` of `source` and
  * the event is related to loading of a tile.
  * @property {Coordinate} [coord] The coordinate of the tile if the event has a `dataType` of `source` and
  * the event is related to loading of a tile.
  * @example
- * // The sourcedata event is an example of MapDataEvent.
- * // Set up an event listener on the map.
- * map.on('sourcedata', function(e) {
- *    if (e.isSourceLoaded) {
- *        // Do something when the source has finished loading
- *    }
- * });
+ * // Example of a MapDataEvent of type "sourcedata":
+ *
+ * {
+ *   type: "sourcedata",
+ *   dataType: "source",
+ *   isSourceLoaded, false,
+ *   source: {...},
+ *   style: {...}
+ *   sourceDataType: "",
```

Comment:
I wasn't sure exactly when this event is emitted. I didn't see it for `sourcedata` type. The current property description suggests maybe it only fires for certain types in certain circumstances, but I wasn't sure exactly what that combo was.

Reply:
Thanks Dan. Hopefully the GL JS team can help us present a clear example of this event.

Reply:
@ryanhamley 

Would this be an appropriate example of a MapDataEvent?

```
{
  dataType: "source",
  isSourceLoaded: false,
  source: {
    type: "vector",
    url: "mapbox://mapbox.mapbox-streets-v8,mapbox.mapbox-terrain-v2"
  },
  sourceDataType: "visibility",
  sourceId: "composite",
  style: {...},
  target: [circular object Object],
  type: "sourcedata"
}
```
Here's a JSFiddle logging such an event: https://jsfiddle.net/domlet/gk872ovy/10/